packages/react-scripts/config/webpackDevServer.config.js

```diff
@@ -22,8 +22,12 @@ const sockPath = process.env.WDS_SOCKET_PATH; // default: '/ws'
 const sockPort = process.env.WDS_SOCKET_PORT;
 
 module.exports = function (proxy, allowedHost) {
-  const disableFirewall =
-    !proxy || process.env.DANGEROUSLY_DISABLE_HOST_CHECK === 'true';
+  let allowedHosts;
+  if (!proxy || process.env.DANGEROUSLY_DISABLE_HOST_CHECK === 'true') {
+    allowedHosts = 'all';
+  } else if (allowedHost != null) {
+    allowedHosts = [allowedHost];
+  }
```

Comment:
I think a more readable way to code it could be something like
```
const disableFirewall = !proxy || process.env.DANGEROUSLY_DISABLE_HOST_CHECK === 'true';
return {
    ... 
    allowedHosts: disableFirewall ? 'all' : allowedHost ? [allowedHost] : undefined,
    ...
};
```
although I didn't like this explicit *undefined* so much. What do you guys think?